fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexScrubbingToolsMissing.java

```diff
@@ -137,59 +125,78 @@ public CompletableFuture<Issue> handleOneItem(final FDBRecordStore store, final
                 });
     }
 
-    public CompletableFuture<Pair<MissingIndexReason, Tuple>> detectMissingIndexKeys(FDBStoredRecord<Message> rec) {
-        // return the first missing (if any).
+    @SuppressWarnings("PMD.CloseResource")
+    private CompletableFuture<Pair<MissingIndexReason, Tuple>> detectMissingIndexKeys(final FDBRecordStore store, FDBStoredRecord<Message> rec) {
```

Comment:
I didn't notice until I went to look at the similarity between this and `ValueIndexScrubbingToolsMissing`, but you don't check the index filter before saving. I think that means you will have false-negatives if any index entries are filtered.
It's probably worth adding a boolean field to `MyParentRecord`, and adding an additional parameter to `saveRecords` to save filtered out records, and change the index definition to filter out anything with that field set to `true` (maybe like `isHidden`)

Reply:
https://github.com/FoundationDB/fdb-record-layer/issues/3065